clients/lsp-eslint.el

```diff
@@ -295,6 +295,10 @@ to allow or deny it.")
              (lsp--persist lsp-eslint-library-choices-file lsp-eslint--stored-libraries)))
          (funcall callback (car option)))))))
 
+(defun lsp-eslint--probe-failed (_workspace _message)
+  "Called when the server detects a misconfiguration in ESLint."
+  (user-error "ESLint is not configured correctly. Please ensure your eslintrc is set up for the languages you are using"))
```

Comment:
given the fact that this happens in async code, I think that lsp--error will be a better fit here.